ecl/hql/hqlgram2.cpp

```diff
@@ -12398,17 +12399,20 @@ void parseAttribute(IHqlScope * scope, IFileContents * contents, HqlLookupContex
     OwnedHqlExpr parsed = scope->lookupSymbol(name, LSFsharedOK|LSFnoreport, ctx);
     bool canCache = parsed && (parsed->getOperator() != no_forwardscope) && !alreadySimplified;
     bool isMacro = parsed && parsed->isMacro();
-    OwnedHqlExpr simplified;
     if (canCache && (ctx.syntaxChecking() || ctx.hasCacheLocation()))
     {
-        simplified.setown(createSimplifiedDefinition(parsed));
+        OwnedHqlExpr simplified = createSimplifiedDefinition(parsed);
```

Comment:
These stats aren't really providing the information that would be useful.  The key questions I would like to ask are:

i) what proportion of attributes can be simplified.
ii) what proportion of attributes are coming from the cache instead of the definition.

This could be improved by moving AttributesNotSimplified to the else branch of if (!simplified), but better would be a count of the number of the total number of attributes processed (and delete the stat for the number of attributes not simplified).

I'm not sure that the cache attribs created is very useful (what question does it help you answer), but happy for it to remain.